docs/dev/packages.rst

```diff
@@ -3,6 +3,72 @@ Package development
 
 Gluon packages are OpenWrt packages and follow the same rules described at https://openwrt.org/docs/guide-developer/packages.
 
+Development workflow
+====================
+
+When you are developing packages, it often happens that you iteratively want to deploy
+and verify the state your development. There are two ways to verify your changes:
+
+1) One way is to rebuild the complete firmware, flash it, configure it and verify your
+   development then. This usually takes at least a few minutes to get your changes
+   working so you can test them. Especially if you iterate a lot, this becomes tedious.
+2) Another way is to rebuild only the package you are currently working on and
+   to deploy this package to your test system. Here not even a reboot is required.
+   This makes iterating relatively fast. Your test system could be real hardware or
+   even a qemu in most cases.
+
+Gluon provides scripts to enhance workflow 2):
+
+.. code-block:: shell
+
+  # start a local qemu instance
+  contrib/run_qemu.sh output/images/factory/[...].img
+
+  # do your changes in the file you want to patch
+  vi package/gluon-ebtables/files/etc/init.d/gluon-ebtables
+
+  # rebuild and push the package
+  contrib/push_pkg.sh package/gluon-ebtables/
+
+  # test your changes
+  ...
+
+  # do more changes
+  ...
+
+  # rebuild and push the package
+  contrib/push_pkg.sh package/gluon-ebtables/
+
+  # test your changes
+  ...
+
+  (and so on...)
+
+  # see help of the script for more information
+  contrib/push_pkg.sh -h
+  ...
+
+Implementation details of ``push_pkg.sh``:
+
+* Currently this is based on ssh/scp.
+* Opkg is used to install/update the packages in the remote machine.
+* It does neither build nor push dependencies of the packages automatically.
+* It will not override /etc/config with package defaults by default. (See the cli switch ``-P``).
+
+Benefits:
+
+* This works with compiled and non-compiled packages.
+* This works with native OpenWrt and Gluon packages.
+* This even performs the ``check_site.lua`` checks as they are integrated as post_install scripts into the openwrt package.
+* It works for all architectures/targets.
```

Comment:
Except `run_qemu.sh` uses KVM on x86_64. The relationship between `run_qemu.sh` and developing on actual hardware isn't clear from this.